util/create-attribution-doc.sh

```diff
@@ -29,23 +29,80 @@ final_license_file=THIRD-PARTY-LICENSES.txt
 # Create a list of aws cdk Sub-packages we want to ignore 
 aws_cdk_ignore_subpackages=$(pip list | grep cdk | awk '{print $1}') 
 aws_cdk_version=$(pip list | grep cdk | awk '{print $2}'| head -n 1)
+
+packaging_version=$(pip list | grep packaging | awk '{print $2}')
+exception_group_version=$(pip list | grep exceptiongroup | awk '{print $2}')
+idna_version=$(pip list | grep idna | awk '{print $2}')
+typing_extentions_version=$(pip list | grep typing_extensions | awk '{print $2}')
 # Create a pip License document
-pip-licenses -i $aws_cdk_ignore_subpackages aws-parallelcluster pip-licenses --format=plain-vertical --with-license-file --with-urls --no-license-path --with-authors --output-file=$final_license_file
+pip-licenses -i $aws_cdk_ignore_subpackages aws-parallelcluster pip-licenses packaging pyinstaller-hooks-contrib pyinstaller exceptiongroup idna typing_extensions certifi --format=plain-vertical --with-license-file --with-urls --no-license-path --with-authors --output-file=$final_license_file
 
 
 #Getting python version
 cpy_version=$(python -V |  grep -Eo '([0-9]+)(\.?[0-9]+)' | head -1) 
 
 
-# Appending aws-cdk and Python License to the THIRD-PARTY-LICENSES file
-# Adding a header for aws-cdk version and License URL to THIRD-PARTY-LICENSES file
-echo "aws-cdk $aws_cdk_version; https://raw.githubusercontent.com/aws/aws-cdk/main/LICENSE" >> $final_license_file
-# Appending aws-cdk License to THIRD-PARTY-LICENSES file
-curl https://raw.githubusercontent.com/aws/aws-cdk/main/LICENSE >> $final_license_file
-# Adding a header for Python version and License URL to THIRD-PARTY-LICENSES file
-echo -e "\nPython $cpy_version; https://raw.githubusercontent.com/python/cpython/$cpy_version/LICENSE " >> $final_license_file
-# Appending Python License to THIRD-PARTY-LICENSES file
-curl https://raw.githubusercontent.com/python/cpython/$cpy_version/LICENSE >> $final_license_file
+# Function to Append Licenses of different Packages to the THIRD-PARTY-LICENSES file
+append_to_final_license_file(){
```

Comment:
so is this function not actually adding the content of the license file, just the URL?  Don't we need to include the full license text?

Reply:
I think we should be specific about this in the name about what exactly we are appending

Reply:
Its appending a header which is Package Name, Package Version, License Type , URL for package to the final License file as well as the whole License file. I have included comments inside the function